entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md

```diff
@@ -1184,3 +1129,532 @@ dotnet new ef-templates
+
+Knowing which interfaces to implement can be tricky, since configuration made to the model at one point may be changed or removed at a later point. For example, a key may be created by convention, but then later replaced when a different key is configured explicitly.
+
+Let's make this a bit more concrete by making a first attempt at implementing the discriminator-length convention:
+
+<!--
+public class DiscriminatorLengthConvention1 : IEntityTypeBaseTypeChangedConvention
+{
+    public void ProcessEntityTypeBaseTypeChanged(
+        IConventionEntityTypeBuilder entityTypeBuilder,
+        IConventionEntityType? newBaseType,
+        IConventionEntityType? oldBaseType,
+        IConventionContext<IConventionEntityType> context)
+    {
+        var discriminatorProperty = entityTypeBuilder.Metadata.FindDiscriminatorProperty();
+        if (discriminatorProperty != null
+            && discriminatorProperty.ClrType == typeof(string))
+        {
+            discriminatorProperty.Builder.HasMaxLength(24);
+        }
+    }
+}
+-->
+[!code-csharp[DiscriminatorLengthConvention1](../../../../samples/core/Miscellaneous/NewInEFCore7/ModelBuildingConventionsSample.cs?name=DiscriminatorLengthConvention1)]
+
+This convention implements <xref:Microsoft.EntityFrameworkCore.Metadata.Conventions.IEntityTypeBaseTypeChangedConvention>, which means it will be triggered whenever the mapped inheritance hierarchy for an entity type is changed. The convention then finds and configures the string discriminator property for the hierarchy.
+
+This convention is then used by calling `Add` in `ConfigureConventions`:
+
+```csharp
+protected override void ConfigureConventions(ModelConfigurationBuilder configurationBuilder)
+{
+    configurationBuilder.Conventions.Add(_ =>  new DiscriminatorLengthConvention1());
+}
+```
+
+> [!TIP]
+> Rather than adding an instance of the convention directly, the `Add` method accepts a factory for creating instances of the convention. This allows the convention to use dependencies from the EF Core internal service provider. Since this convention has no dependencies, the service provider parameter is named `_`, indicating that it is never used.
+
+Building the model and looking at the `Post` entity type shows that this has worked--the discriminator property is now configured to with a maximum length of 24:
+
+```text
+ Discriminator (no field, string) Shadow Required AfterSave:Throw MaxLength(24)
+```
+
+But what happens if we now explicitly configure a different discriminator property? For example:
+
+```csharp
+modelBuilder.Entity<Post>()
+    .HasDiscriminator<string>("PostTypeDiscriminator")
+    .HasValue<Post>("Post")
+    .HasValue<FeaturedPost>("Featured");
+```
+
+Looking at the model, we find that the discriminator length is no longer configured!
```

Comment:
How are you looking at the model?

Reply:
Debug view (ToDebigString) but we don't have any docs for that yet.

Reply:
Right, it would be helpful to mention it